indexer/cell_coverer.hpp

```diff
@@ -2,54 +2,58 @@
 
 #include "indexer/cell_id.hpp"
 
+#include "base/buffer_vector.hpp"
+
 #include "std/queue.hpp"
 #include "std/vector.hpp"
 
 // TODO: Move neccessary functions to geometry/covering_utils.hpp and delete this file.
 
+constexpr int SPLIT_RECT_CELLS_COUNT = 512;
+
 template <typename BoundsT, typename CellIdT>
-inline void SplitRectCell(CellIdT id,
-                          double minX, double minY,
-                          double maxX, double maxY,
-                          vector<CellIdT> & result)
+inline int SplitRectCell(CellIdT const & id, m2::RectD const & rect,
```

Comment:
I suggest you to fill buffer_vector instead of array and use push_backs to fill it. Same thing, but you don't need to return int.